.buildkite/pipelines/pull_request/base.yml

```diff
@@ -19,6 +19,14 @@ steps:
         - exit_status: '-1'
           limit: 3
 
+  - command: .buildkite/scripts/steps/archive_so_migration_snapshot.sh target/plugin_so_types_snapshot.json
+    label: 'Extract Saved Object migration plugin types'
+    depends_on: build
```

Comment:
maybe it doesn't need to depend on the build step, because every step runs on a clean slate, this step will build its own dev-kibana either way